tests/phpunit/api/v3/OptionValueCHTest.php

```diff
@@ -192,4 +200,113 @@ public function testCreateOptionValueChange() {
+    // ensure that the value of financial type is not changed
+    $this->assertEquals($this->fund['id'], $contribution['financial_type_id']);
+    $lineItem = $this->callAPISuccess('lineItem', 'get', [
+      'contribution_id' => $contributionID,
+      'sequential' => 1,
+    ]);
+    // ensure that one line-item is present
+    $this->assertEquals(1, $lineItem['count']);
+    // ensure that associated line-item has correct financial item linked to it
+    $this->assertEquals($this->fund['id'], $lineitem[0]['financial_type_id']);
+    $totalFinancialItem = $this->callAPISuccess('FinancialItem', 'getcount', [
+      'entity_table' => 'civicrm_line_item',
+      'entity_id' => $lineItem['id'],
+    ]);
+    $this->assertEquals(1, $totalFinancialItem);
+
+    // delete created values
+    $this->callAPISuccess('contribution', 'delete', ['id' => $contributionID]);
+    $this->callAPISuccess('OptionValue', 'delete', ['id' => $chFund['id']]);
+    $updatedMap = $this->callAPISuccess('OptionValueCH', 'get', []);
+    $this->assertEmpty($updatedMap['values']);
+  }
+
+  public function testContributionOnOptionvalueCHFundChange() {
+      $chFund = $this->callAPISuccess('OptionValue', 'create', [
+      'label' => 'Test Created CH Fund 1',
+      'option_group_id' => 'ch_fund',
+      'value' => 'CH+99999',
+    ]);
+
+    $contributionID = $this->callAPISuccess('CHContribution', 'create', [
+      'receive_date' => date('Ymd'),
+      'total_amount' => 100.00,
+      'payment_instrument_id' => 1,
+      'source' => 'SSF',
+      'contribution_status_id' => 1,
+      'ch_fund' => 'CH+99999',
+      'contact_id' => $this->individualID,
+    ])['id'];
+    $contribution = $this->callAPISuccess('Contribution', 'getsingle', [
+      'id' => $contributionID,
+    ]);
+    // ensure that contribution Fund is assigned to 'Unassigned CH Fund'
+    $this->assertEquals($this->unallocatedFund['id'], $contribution['financial_type_id']);
+
+    $result = CRM_Core_DAO::executeQuery("SELECT * FROM civicrm_ch_contribution_batch ")->fetchAll();
+    // ensure that the queue is empty before updating the mapping by changing the fund value
+    $this->assertEmpty($result);
+
+    $chFundMap = $this->callAPISuccess('OptionValueCH', 'getsingle', ['value' => 'CH+99999']);
+    // change Fund value in the mapping
+    $this->callAPISuccess('OptionValueCH', 'create', ['fund' => $this->fund['id'], 'id' => $chFundMap['id']]);
+
+    // check there is entry in contribution batch table to mark this change in fund value
+    $result = CRM_Core_DAO::executeQuery("SELECT * FROM civicrm_ch_contribution_batch ")->fetchAll();
+    $this->assertNotEmpty($result);
+    $this->assertEquals(1, count($result));
+    // ensure the contribution which needs to be updated is present in the queue
+    $this->assertEquals($contributionID, $result[0]['contribution_id']);
+    $this->assertEquals($this->fund['id'], $result[0]['fund']);
```

Comment:
@monishdeb so if the CHFund option value changes would this have any impact on the batch process?

Reply:
No it wouldn't because CRM_Core_BAO_CustomOption::updateValue() is taking care of updating the CH Fund custom field values of all contributions. So we don't want to have process contributions again. In case of Fund change we need to change the contribution's fund via contribution.create API which is an expensive operation and thus need to be processed in batches.